src/components/select/filterable-select/filterable-select.spec.js

```diff
@@ -1004,6 +1033,40 @@ describe("FilterableSelect", () => {
           expect(onOpenFn).toHaveBeenCalled();
         });
 
+        it("then it should not be invoked when the component rerenders", () => {
+          const Component = () => {
+            const [callCounter, setCallCounter] = useState(0);
+
+            return (
+              <>
+                <div id="call-counter">{callCounter}</div>
+                <FilterableSelect
+                  name="testSelect"
+                  id="testSelect"
+                  openOnFocus
+                  onOpen={() => {
+                    if (callCounter < 2) {
+                      setCallCounter(callCounter + 1);
+                    }
+                  }}
+                >
+                  <Option value="opt1" text="red" />
+                  <Option value="opt2" text="green" />
+                  <Option value="opt3" text="blue" />
+                  <Option value="opt4" text="black" />
+                </FilterableSelect>
+              </>
+            );
+          };
+
+          wrapper = mount(<Component />);
+          expect(wrapper.find("#call-counter").text()).toBe("0");
+          wrapper.find("input").simulate("focus");
+          expect(wrapper.find("#call-counter").text()).toBe("1");
+          wrapper.setProps({});
+          expect(wrapper.find("#call-counter").text()).toBe("1");
```

Comment:
similar here to my question about the other test - why use state and DOM updates to track the number of calls when it can be done directly with `jest.fn()`?